server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java

```diff
@@ -202,6 +203,23 @@ private String getLocationFamily(LocationType type) {
     return this;
   }
 
+  @Override
+  public Ample.TabletMutator putSuspension(Ample.TServer tServer, long suspensionTime) {
+    Preconditions.checkState(updatesEnabled, "Cannot make updates after calling mutate.");
+    mutation.put(SuspendLocationColumn.SUSPEND_COLUMN.getColumnFamily(),
+        SuspendLocationColumn.SUSPEND_COLUMN.getColumnQualifier(),
+        new Value(tServer + "|" + suspensionTime));
```

Comment:
This replaces `SuspendingTServer.toValue()`, which does not appear to be used any longer. The old method stored a `HostAndPort.toString()`, but this stores `Ample.TServer.toString()`. Are these identical?

The serialization for `toValue` should probably remain in the `SuspendingTServer` class, since that's where the corresponding `fromValue` method is. So, I recommend changing `SuspendingTServer.toValue` that is called from here, so if the serialization changes, it can be changed in both `toValue` and `fromValue` at the same time.

Reply:
Thanks for pointing that out, that was something I didn't notice. I replaced with ` new Value(tServer.getLocation() + "|" + suspensionTime))` which more accurately matches `HostAndPort.toString()`. This also solves the issue I was facing with `MasterRepairsDualAssignmentIT` so I no longer need the if statement I added in the `HostAndPort` file.

I don't fully understand your second paragraph. Are you suggesting I use the `SuspendingTServer.fromValue()` function in this mutation instead of doing the `new Value(tServer.getLocation() + "|" + suspensionTime))`?

Reply:
> I don't fully understand your second paragraph. Are you suggesting I use the `SuspendingTServer.fromValue()` function in this mutation instead of doing the `new Value(tServer.getLocation() + "|" + suspensionTime))`?

Prior to your changes, the `SuspendingTServer.toValue()` was responsible for creating this `new Value`. I'm suggesting that we leave that class responsible for creating the `new Value`,  so it can live alongside that class' existing `fromValue` method. In order to do that, you would need to turn it into a static method, so you can use it in your code.

In other words, this code should change to something like:

```suggestion
        SuspendingTServer.toValue(tServer, suspensionTime);
```

And then that method should look something like:

```java
    public static Value toValue(HostAndPort tServer, long suspensionTime) {
      return new Value(tServer + "|" + suspensionTime);
    }
```

I probably don't have the types correct in this code, but something like that is what I'm suggesting.